app/controllers/v0/sign_in_controller.rb

```diff
@@ -90,7 +90,8 @@ def token
       validated_credential = SignIn::CodeValidator.new(code: code,
                                                        code_verifier: code_verifier,
                                                        grant_type: grant_type).perform
-      session_container = SignIn::SessionCreator.new(validated_credential: validated_credential).perform
+      session_container = SignIn::SessionCreator.new(validated_credential: validated_credential,
+                                                     request_ip: request.ip).perform
```

Comment:
I wonder if a slightly better spot for this would be in the `SignIn::UserCreator.new`, since that is when the `User` object itself gets created. Seems like it'd be slightly cleaner in there, as an extra line in the `create_authenticated_user` function

Reply:
`UserCreator` gets called in `create_login_code` at the end of the `/callback` route - it was my impression that this story was supposed to be setting the user fingerprint during the session creation in `/token` as opposed to earlier.

Reply:
I feel like `/callback` is a slightly 'safer' place to put it since it will have to be the device of the the user who entered credential information, and it seems cleaner since it's a place we're already messing with the user model. So I feel like it's a win-win to put it in there